### PR TITLE
plugins.twitch: fix call_subdomain

### PR DESCRIPTION
**Pull request by EnterGin**
fixes https://github.com/streamlink/streamlink/issues/2751

### CODE REVIEW

File: src/streamlink/plugins/twitch.py [hunk 1]
```diff
@@ -306,7 +306,11 @@ def call(self, path, format="json", schema=None, private=False, **extra_params):
...
         subdomain_buffer = self.subdomain
         self.subdomain = subdomain
-        response = self.call(path, format=format, schema=schema, **extra_params)
+        try:
+            response = self.call(path, format=format, schema=schema, **extra_params)
+        except PluginError:
+            self.subdomain = subdomain_buffer
+            raise
         self.subdomain = subdomain_buffer
         return response
```
Comment (bastimeyer) [inline src/streamlink/plugins/twitch.py:315]:
```suggestion
        try:
            return self.call(path, format=format, schema=schema, **extra_params)
        finally:
            self.subdomain = subdomain_buffer
```

**Comment by EnterGin** [comment]
> That's not the diff I suggested. You are writing duplicate code.

Oh, you about removing subdomain at all?

**Comment by bastimeyer** [comment]
> diff

https://github.com/streamlink/streamlink/pull/2958#discussion_r423973855

**Comment by EnterGin** [comment]
But it's actually changed
image

**Comment by EnterGin** [comment]
OK, I got it, sorry.

**Pull request merged**